src/sentry/api/endpoints/project_metrics.py

```diff
@@ -1,36 +1,70 @@
 from rest_framework.exceptions import ParseError
 from rest_framework.response import Response
 
-from sentry import features
 from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
 from sentry.snuba.metrics import DATA_SOURCE, InvalidField, InvalidParams, QueryDefinition
 
 
 class ProjectMetricsEndpoint(ProjectEndpoint):
-    """ Get metric name, type, unit and tag names """
+    """ Get metric name, available operations and the metric unit """
 
     def get(self, request, project):
-
-        if not features.has("organizations:metrics", project.organization, actor=request.user):
-            return Response(status=404)
-
         metrics = DATA_SOURCE.get_metrics(project)
         return Response(metrics, status=200)
 
 
+class ProjectMetricDetailsEndpoint(ProjectEndpoint):
+    """ Get metric name, available operations, metric unit and available tags """
+
+    def get(self, request, project, metric_name):
+        try:
+            metric = DATA_SOURCE.get_single_metric(project, metric_name)
+        except InvalidParams:
+            raise ResourceDoesNotExist(detail=f"metric '{metric_name}'")
+
+        return Response(metric, status=200)
+
+
 class ProjectMetricsTagsEndpoint(ProjectEndpoint):
-    """ Get all existing tag values for a metric """
+    """Get list of tag names for this project
+
+    If the ``metric`` query param is provided, only tags for a certain metric
+    are provided.
 
-    def get(self, request, project, metric_name, tag_name):
+    If the ``metric`` query param is provided more than once, the *intersection*
+    of available tags is used.
 
-        if not features.has("organizations:metrics", project.organization, actor=request.user):
-            return Response(status=404)
+    """
+
+    def get(self, request, project):
+
+        metric_names = request.GET.getlist("metric") or None
 
         try:
-            tag_values = DATA_SOURCE.get_tag_values(project, metric_name, tag_name)
+            tag_names = DATA_SOURCE.get_tag_names(project, metric_names)
         except InvalidParams as exc:
             raise (ParseError(detail=str(exc)))
 
+        return Response(tag_names, status=200)
+
+
+class ProjectMetricsTagDetailsEndpoint(ProjectEndpoint):
+    """ Get all existing tag values for a metric """
+
+    def get(self, request, project, tag_name):
+
+        metric_names = request.GET.getlist("metric") or None
+
+        try:
+            tag_values = DATA_SOURCE.get_tag_values(project, tag_name, metric_names)
+        except InvalidParams as exc:
+            msg = str(exc)
+            if "Unknown tag" in msg:
```

Comment:
When this endpoint has real data would this be a separate error type?

Reply:
Yes, I've added a comment so we don't forget.